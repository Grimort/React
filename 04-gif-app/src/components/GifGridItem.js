import React from "react";

const GifGridItem = ({ id, title, url }) => {
  return (
    <div className="card animate__animated animate__bounce animate__fadeInDown">
      <p>{title}</p>
      <img src={url.url} alt={title}></img>
    </div>
  );
};

export default GifGridItem;
