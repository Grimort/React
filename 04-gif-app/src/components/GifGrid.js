import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn" key={category}>
        {category}
      </h3>
      <div className="card-grid animate__animated animate__bounce animate__fadeIn">
        {loading && (
          <p className="animate__animated animate__flash">Loading...</p>
        )}
        {images.map((img) => (
          <GifGridItem key={img.id} url={img.url.url} title={img.title} />
        ))}
      </div>
    </>
  );
};
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
