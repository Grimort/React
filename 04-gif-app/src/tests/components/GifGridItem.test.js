import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Testing GifGridItem component", () => {
  const title = "Title";
  const url = "https://localhost.img.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test("should to show the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should have a <p></p> with the title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });
  test("The img should have the url and alt like the props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("should have the animation class", () => {
    const div = wrapper.find("div");
    expect(div.prop("className")).toContain("animate__fadeInDown");
  });
});
