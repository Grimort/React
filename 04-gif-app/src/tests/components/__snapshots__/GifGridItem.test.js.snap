// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem component should to show the component correctly 1`] = `
<div
  className="card animate__animated animate__bounce animate__fadeInDown"
>
  <p>
    Title
  </p>
  <img
    alt="Title"
    src="https://localhost.img.jpg"
  />
</div>
`;
